src/mapping.ts

```diff
@@ -1,410 +1,477 @@
-  // if (option === null) {
-  //   option = new Option(event.params.optionId.toString());
-  // }
-
-  let exerciseAssetAddress = option.exerciseAsset as string;
-  let exercisePriceUSD = getTokenPriceUSD(exerciseAssetAddress);
-  let exerciseAmount = (option.exerciseAmount as BigInt)
-    .toBigDecimal()
-    .div(
-      exponentToBigDecimal(
-        BigInt.fromI64(
-          ERC20.bind(Address.fromString(exerciseAssetAddress)).decimals()
-        )
-      )
-    );
-  let exerciseValueUSD = exercisePriceUSD
-    .times(exerciseAmount)
-    .times(event.params.amount.toBigDecimal());
-
-  let underlyingAssetAddress = option.underlyingAsset as string;
-  let underlyingPriceUSD = getTokenPriceUSD(underlyingAssetAddress);
-  let underlyingAmount = (option.underlyingAmount as BigInt)
-    .toBigDecimal()
-    .div(
-      exponentToBigDecimal(
-        BigInt.fromI64(
-          ERC20.bind(Address.fromString(underlyingAssetAddress)).decimals()
-        )
-      )
-    );
-
-  let underlyingValueUSD = underlyingPriceUSD
-    .times(underlyingAmount)
-    .times(event.params.amount.toBigDecimal());
-
-  let contract = fetchERC1155(event.address);
-
-  // Update TVL to reflect the value of the exercise tokens being transfered in
-  // and underlying tokens being transfered out
-  contract.totalValueLockedUSD = contract.totalValueLockedUSD
-    .plus(exerciseValueUSD)
-    .minus(underlyingValueUSD);
-
-  contract.save();
-
-  let underlyingAsset = loadOrInitializeToken(
-    Address.fromString(underlyingAssetAddress as string)
+  // handle daily metrics
+  const tokenDaily = fetchDailyTokenMetrics(
+    event.params.asset.toHexString(),
+    event.block.timestamp
   );
-  underlyingAsset.totalValueLocked = underlyingAsset.totalValueLocked.minus(
-    underlyingAmount
+  tokenDaily.notionalVolFeesSwept = tokenDaily.notionalVolFeesSwept.plus(
+    event.params.amount
   );
-  underlyingAsset.totalValueLockedUSD = underlyingAsset.totalValueLockedUSD.minus(
-    underlyingValueUSD
+  tokenDaily.notionalVolFeesSweptUSD = tokenDaily.notionalVolFeesSweptUSD.plus(
+    feeValueUSD
   );
-  underlyingAsset.save();
+  tokenDaily.save();
 
-  let exerciseAsset = loadOrInitializeToken(
-    Address.fromString(exerciseAssetAddress as string)
-  );
-  exerciseAsset.totalValueLocked = exerciseAsset.totalValueLocked.plus(
-    exerciseAmount
-  );
-  exerciseAsset.totalValueLockedUSD = exerciseAsset.totalValueLockedUSD.plus(
-    exerciseValueUSD
+  const dailyOSEMetrics = fetchDailyOSEMetrics(event.block.timestamp);
+  dailyOSEMetrics.notionalVolFeesSweptUSD = dailyOSEMetrics.notionalVolFeesSweptUSD.plus(
+    feeValueUSD
   );
-  exerciseAsset.save();
-
-  let dayData = updateValoremDayData(event);
-
-  dayData.volumeUSD = dayData.volumeUSD.plus(underlyingValueUSD);
+  dailyOSEMetrics.save();
+}
 
-  dayData.save();
+// https://github.com/OpenZeppelin/openzeppelin-subgraphs
```

Comment:
Standardize here and elsewhere how we include and credit code+license type. This doesn't match the first few.

Reply:
👍

Reply:
done in https://github.com/valorem-labs-inc/valorem-subgraph/commit/ecb0958c55a60effddb94cc5d773b72580758013